_posts/2020-02-15-pmemkv-bindings.md

```diff
@@ -0,0 +1,155 @@
+---
+title: Language bindings for pmemkv
+author: lukaszstolarczuk
+layout: post
+identifier: pmemkv-bindings
+---
+
+Along with [pmemkv][pmemkv_pmem_io] written in C/C++ there are a bunch of
+bindings for high-level languages. Currently: Java (with JNI), Node.js, Python and Ruby.
+
+Picture below properly summarize architecture and software stack of pmemkv and its bindings.
+![pmemkv_bindings_image](/assets/pmemkv_bindings.png)
+
+Always, the most current updates about bindings (and pmemkv in general) can be found on
+[pmemkv's README file][pmemkv_bindings_readme].
+
+### Common for bindings
+
+There are few common characteristics regarding all bindings:
+
+#### Work with libpmemkv 1.0
+
+All of our currently released bindings work properly with libpmemkv 1.0. Unfortunately their API
+may not be fully functionally equal to the native C/C++ API. These **bindings, which are
+in version 1.0** deliver API in accordance with **pmemkv 1.0**. In general, pmemkv's binding
+released in version X.Y is (always) functionally greater or equal to pmemkv in the same X.Y version.
+Additionally, if a binding's version is prior to 1.0 (any 0.x version), it should be considered
+an experimental implementation.
+
+Taking into consideration that libpmemkv is developing in the fast pace, there are always
+some gaps in the current versions of bindings' implementations. Even more, each binding
+is developed in its own pace, so the best way to check compatibility with pmemkv
+is to read changelog of the latest release of a specific binding.
+
+#### Handling pmemkv statuses
+
+Every language supports handling statuses returned from pmemkv's API. Each has its own way,
+either translating each status into numeric value or wrapping them into exceptions. Bindings
+using exceptions do not use status "OK", just returns nothing or a numeric value
+(of `PMEMKV_STATUS_OK`) if no error occurred. Return value differs also for various functions -
+to see details read binding's documentation or pmemkv's man pages.
+
+#### Not always direct access to data
+
+**Except in Python** (and soon [in Node.js][nodejs_pr]), our bindings do not (yet)
+provide direct access to the data stored in the Persistent Memory.
+They need to use buffer to copy the data to.
+
+#### Examples, tests and CI
+
+Each repository is equipped with basic, needed features like:
+* examples, showing basic library usage,
+* tests, verifying current implementation,
+* Continuous Integration system, to check all incoming changes.
+
+#### Packaging specific for language
+
+Each language is different, so we try our best to use packaging and installation systems
+native to (or one of the most common for) each binding's language.
+
+#### Contributions are welcome
+
+As all of pmem's organization repositories, the ones of pmemkv-bindings are also
```

Comment:
Well, it's github, so unless we used a non-free license, it's a given that projects in pmem organization are open source.
I'd rephrase this to say that "pmemkv is BSD, so we welcome contributions."